### PR TITLE
Cleanup of poms

### PR DESCRIPTION
**Pull request by arjantijms**
* Formatting
* Removing unused/unnecessary plug-ins
* Updating leftover terms (JavaServer to Jakarta Server, ...)

Signed-off-by: arjantijms <arjan.tijms@gmail.com>

### CODE REVIEW

File: impl/pom.xml [hunk 1]
```diff
@@ -70,51 +45,124 @@
...
-                        <phase>process-classes</phase>
+                        <id>enforce-maven</id>
                         <goals>
-                            <goal>manifest</goal>
+                            <goal>enforce</goal>
                         </goals>
+                        <configuration>
+                            <rules>
+                                <requireMavenVersion>
+                                    <version>3.3.0</version>
```
Comment (joakime) [inline impl/pom.xml:157]: This statement here, doesn't jive with the comment above (on line 143).

File: impl/pom.xml [hunk 2]
```diff
@@ -70,51 +45,124 @@
...
+                                </requireMavenVersion>
+                            </rules>
+                        </configuration>
                     </execution>
                 </executions>
             </plugin>
+            
+            <!-- Restricts the Java version to 1.8 -->
```
Comment (joakime) [inline impl/pom.xml:165]: Since this is the `/impl` pom, shouldn't this be Java 11?

File: pom.xml [hunk 3]
```diff
@@ -60,4 +47,16 @@
         </license>
     </licenses>
 
+    <modules>
+        <module>api</module>
+        <module>spec</module>
+        <module>impl</module>
```
Comment (joakime) [inline pom.xml:53]: The `jdsol-spec` modules should probably also be on this module list

File: impl/pom.xml [hunk 4]
```diff
@@ -70,51 +45,124 @@
...
+        
+        <!-- Other dependencies -->
+        <dependency>
+            <groupId>org.apache.ant</groupId>
+            <artifactId>ant</artifactId>
+            <version>1.10.7</version>
+        </dependency>
+        <dependency>
+          <groupId>eclipse</groupId>
+          <artifactId>jdtcore</artifactId>
```
Comment (joakime) [inline impl/pom.xml:118]: Odd indentation here.

I think we should just settle on 2 spaces, no tabs and be done with it.
That's the default on both Eclipse IDE and IntelliJ for XML anyway.

File: impl/pom.xml (hunk 1 above)
Comment (arjantijms) [reply impl/pom.xml:157]: Good catch, I'll update that!

File: impl/pom.xml (hunk 2 above)
Comment (arjantijms) [reply impl/pom.xml:165]: Mark just asked for it to remain on 8, and just make sure it compiles/works on 11. For Jakarta EE 9 I think that's fair.

File: impl/pom.xml (hunk 4 above)
Comment (arjantijms) [reply impl/pom.xml:118]: I think the default for Jakarta is 4 spaces, indeed no tabs. I'll update it, thanks again!

File: pom.xml (hunk 3 above)
Comment (arjantijms) [reply pom.xml:54]: I could add that in another PR. This PR is mostly about the formatting and plug-ins etc.

**Comment by arjantijms** [comment]
@joakime I processed your feedback, let me know what you think ;)

**Pull request merged**